--- a/astropy/modeling/separable.py
+++ b/astropy/modeling/separable.py
@@ -207,6 +207,7 @@ def _coord_matrix(model, pos, noutp):
 
     Examples
     --------
+    >>> from astropy.modeling.models import Shift
     >>> _coord_matrix(Shift(1), 'left', 1)
     array([[1, 0]]))
 
@@ -217,12 +218,20 @@ def _cstack(left, right):
 
     carr = []
     if isinstance(left, np.ndarray):
         carr.append(left)
+    elif isinstance(left, CompoundModel):
+        # Recursively get the separability for compound models
+        cleft = _separable(left)
+        carr.append(cleft)
     else:
         cleft = _coord_matrix(left, 'left', noutp)
         cleft = np.rollaxis(cleft, -1, -2)
         carr.append(cleft)
+        
     if isinstance(right, np.ndarray):
         carr.append(right)
+    elif isinstance(right, CompoundModel):
+        # Recursively get the separability for compound models
+        cright = _separable(right)
+        carr.append(cright)
     else:
         cright = _coord_matrix(right, 'right', noutp)
         cright = np.rollaxis(cright, -1, -2)